src/components/Home.jsx

```diff
@@ -1,7 +1,44 @@
 import React from 'react';
+import getToken from '../lib/tokens';
+import { Redirect } from 'react-router-dom';
+import { useHistory } from 'react-router-dom';
 
 const Home = () => {
-  return <h1>Home</h1>;
+  // const [hasToken, setHasToken] = useState(false);
+
+  let history = useHistory();
+
+  const Token = function generateToken() {
+    const token = getToken();
+    localStorage.setItem('token', token);
+    history.push('/list-view');
+
+    console.log(token);
+  };
+
+  // function fetchToken() {
+  //   localStorage.getItem('token');
+  //   history.push('/list-view');
+  // }
+
+  const existingToken = localStorage.getItem('token');
+
+  return (
+    <div>
+      <button onClick={Token}>Create a new list</button>
+      {existingToken ? <Redirect from="/" to="/list-view" /> : ''}
```

Comment:
I am Learning a lot about token, I like how you've managed to import the tokens files and be able to use  in your codes. The suggestion I have is that you can loop through those tokens so that your tenary operator condition can work in right way, If  am right . 
<a href="https://www.loom.com/share/9e3b5163db04467f85395a7ab8013baf">
    <p>Cy nr setup newlist by blackbureau · Pull Request #22 · the-collab-lab/tcl-29-smart-shopping-list - Watch Video</p>
    <img style="max-width:300px;" src="https://cdn.loom.com/sessions/thumbnails/9e3b5163db04467f85395a7ab8013baf-with-play.gif">
  </a>
I think that this stack over flow  discussion might help :https://stackoverflow.com/questions/52428879/objects-are-not-valid-as-a-react-child-if-you-meant-to-render-a-collection-of-c